### DISCUSSION TITLE
How can I dynamically add Service groups and Endpoints

### DISCUSSION THREAD

**Discussion by alexandrzamboch** [phax/phoss-smp#231] [General]
Hi,
I would like to dynamically add Endpoints and Service Groups to Phoss-SMP. Can you please advice the best way how to achieve that? I am using the Phoss-SMP which stores Endpoints and in XML files. 

Thank you,
Regards,
Alex

**Comment by phax** [comment]
Hi Alex,
Please refer to the REST-API documentation in the Wiki:
https://github.com/phax/phoss-smp/wiki/REST-API

hth, Philip

**Comment by alexandrzamboch** [comment]
Thank you Philip!

**Reply by alexandrzamboch** [reply]
Hi Philip,

is there a way I can invoke phoss-smp rest api from C#?

Thank you,
Regards,
Alex

**Reply by phax** [reply]
Well sure.
You create some code that creates the necessary XML according to the API interface (see https://github.com/phax/phoss-smp/wiki/REST-API).
Afterwards you use a simple HTTP client to GET/PUT/DELETE the data to the respective interface.
So it's no rocket science, but I am not a .NET guy, so I can't even point you to proper examples.

Something like that looks reasonable: https://learn.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client - of course I don't know if still is still up to date etc.

hth

**Reply by alexandrzamboch** [reply]
Thank you Philip!
